src/tactic/lean_core_docs.lean

```diff
@@ -0,0 +1,650 @@
+add_tactic_doc
+{ name       := "cases",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.cases],
+  tags       := ["core", "basic", "induction"] }
+
+/--
+`cases_matching p` applies the `cases` tactic to a hypothesis `h : type`
+if `type` matches the pattern `p`.
+
+`cases_matching [p_1, ..., p_n]` applies the `cases` tactic to a hypothesis `h : type`
+if `type` matches one of the given patterns.
+
+`cases_matching* p` is a more efficient and compact version of `focus1 { repeat { cases_matching p } }`.
+It is more efficient because the pattern is compiled once.
+
+`casesm` is shorthand for `cases_matching`.
+
+Example: The following tactic destructs all conjunctions and disjunctions in the current context.
+```
+cases_matching* [_ ∨ _, _ ∧ _]
+```
+-/
+add_tactic_doc
+{ name       := "cases_matching / casesm",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.cases_matching, `tactic.interactive.casesm],
+  tags       := ["core", "induction", "context management"] }
+
+/--
+`cases_type I` applies the `cases` tactic to a hypothesis `h : (I ...)`
+`cases_type I_1 ... I_n` applies the `cases` tactic to a hypothesis `h : (I_1 ...)` or ... or `h : (I_n ...)`
+`cases_type* I` is shorthand for `focus1 { repeat { cases_type I } }`
+`cases_type! I` only applies `cases` if the number of resulting subgoals is <= 1.
+
+Example: The following tactic destructs all conjunctions and disjunctions in the current context.
+```
+cases_type* or and
+```
+-/
+add_tactic_doc
+{ name       := "cases_type",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.cases_type],
+  tags       := ["core", "induction", "context management"] }
+
+add_tactic_doc
+{ name       := "change",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.change],
+  tags       := ["core", "basic", "renaming"] }
+
+add_tactic_doc
+{ name       := "clear",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.clear],
+  tags       := ["core", "context management"] }
+
+/--
+Close goals of the form `n ≠ m` when `n` and `m` have type `nat`, `char`, `string`, `int` or `fin sz`,
+and they are literals. It also closes goals of the form `n < m`, `n > m`, `n ≤ m` and `n ≥ m` for `nat`.
+If the goal is of the form `n = m`, then it tries to close it using reflexivity.
+
+In mathlib, consider using `norm_num` instead for numeric types.
+-/
+add_tactic_doc
+{ name       := "comp_val",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.comp_val],
+  tags       := ["core", "arithmetic"] }
+
+/--
+The `congr` tactic attempts to identify both sides of an equality goal `A = B`,
+leaving as new goals the subterms of `A` and `B` which are not definitionally equal.
+Example: suppose the goal is `x * f y = g w * f z`. Then `congr` will produce two goals:
+`x = g w` and `y = z`.
+
+Note that `congr` can be over-aggressive at times; in mathlib the `congr'` tactic in mathlib
```

Comment:
"in mathlib" is used twice. Is it intentional?

Reply:
No! Thanks.